### PR TITLE
Add NDTreeModel, for N-dimensions KDTreeModel

### PR DESCRIPTION
**Pull request by jq**
@deerzq @hectorgon @yolken

### CODE REVIEW

File: core/src/main/java/com/airbnb/aerosolve/core/models/NDTreeModel.java [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Stack;
+
+/*
+  N-Dimensional KDTreeModel.
+ */
+public class NDTreeModel implements Serializable {
+  private static final long serialVersionUID = -2884260218927875615L;
+  private static final Logger log = LoggerFactory.getLogger(KDTreeModel.class);
```
Comment (spencerde) [inline core/src/main/java/com/airbnb/aerosolve/core/models/NDTreeModel.java]: `NDTreeModel.class` for logger?

File: core/src/main/thrift/NDTree.thrift [hunk 2]
```diff
@@ -0,0 +1,14 @@
+namespace java com.airbnb.aerosolve.core
+
+struct NDTreeNode {
+// coordinateIndex = min/max size is child,
+// coordinateIndex from 0 to min.size()-1 means split in that coordinate
+// similar to KDTreeNode's X_SPLICT, Y_SPLICT
```
Comment (spencerde) [inline core/src/main/thrift/NDTree.thrift]: sp/X_SPLICT/X_SPLIT, Y_SPLICT/Y_SPLIT

**Comment by spencerde** [comment]
Nice! Small comments overall LGTM

**Comment by jq** [comment]
@spencerde @deerzq Thanks for review!! all issues fixed.

**Pull request merged**